synchronize.py

```diff
@@ -0,0 +1,156 @@
+            pickle.dump(creds, token)
+    return creds
+
+def capture_events():
+    events_list = []
+    syncToken = get_token()
+    service = build('calendar', 'v3', credentials=get_credentials())
+    # Call the Calendar API
+    events_result = service.events().list(calendarId=calendarId,
+                                        syncToken=syncToken).execute()
+    events = events_result.get('items', [])
+    events_list.append(events)
+    # We have captured all of the events since the last token, only when the response
+    # returns the nextSyncToken
+    nextSyncToken = events_result.get('nextSyncToken',None)
+    while nextSyncToken is None:
+        nextPageToken = events_result.get('nextPageToken',None)
+        events_result = service.events().list(calendarId=calendarId,
+            syncToken=syncToken, pageToken=nextPageToken).execute()
+        events = events_result.get('items', [])
+        events_list.append(events)
+        nextSyncToken = events_result.get('nextSyncToken',None)
+    store_token(nextSyncToken)
+    return events_list
+
+def interpret_events(list_of_events):
+    events_to_insert = []
+    cancelled_event_ids = []
+    for event_list in list_of_events:
+        for event in event_list:
+            #TODO figure out timezone
+            print(event)
+            if event['status'] in ['confirmed', 'tentative']:
+                event_id = event['id']
+                status = 1
+                label = event['summary']
+                datastr = {'event_id':event_id, 'label':label, 'status':status}
+                if ('recurrence' in event.keys()): datastr['recurrence'] = str(event['recurrence'])
+                if ('recurringEventId' in event.keys()): datastr['recurringEventId'] = event['recurringEventId'] 
+                if 'originalStartTime' in event.keys():
+                    originalStartTime = event['originalStartTime'].get('dateTime', event['originalStartTime'].get('date'))
+                    datastr['originalStartTime'] = originalStartTime
+                    originalTimeZone = event['originalStartTime'].get('timeZone')
+                if event['start'].get('dateTime') != None:
+                    start = event['start'].get('dateTime')
+                    start_time = datetime.strptime(start, '%Y-%m-%dT%H:%M:%S%z')
+                else:
+                    start = event['start'].get('date')
+                    start_time = datetime.strptime(start, '%Y-%m-%d')
+                start_timezone = event['start'].get('timeZone', None)
+                if event['end'].get('dateTime') != None:
+                    end = event['end'].get('dateTime')
+                    end_time = datetime.strptime(end, '%Y-%m-%dT%H:%M:%S%z')
+                else:
+                    end = event['end'].get('date')
+                    end_time = datetime.strptime(end, '%Y-%m-%d')
+                end_timezone = event['end'].get('timeZone', None)
+
+                duration = end_time - start_time
```

Comment:
I'm not sure whether this accounts for the timezone already. The event has a separate timeZone property both for start and end time which specifies the city, which I'm currently not using anywhere. Any advice or references on how you managed the timezones in ActivityWatch would be useful.

Reply:
Tested for the timezone, seems to work ok